### PR TITLE
Drop unsupported Django versions

### LINKED ISSUE #183: Remove Django itself from install_requires
There's an unfortunate side effect of referencing Django itself in setup.py or requirements.txt

It rather unexpectedly results in an upgrade to Django itself if I type: "pip install django-model-utils -U"

And really - your app doesn't 'require' django as much as it's an add-on for Django so not much is lost by removing it. Is anyone likely to install django-model-utils and expect it to install Django for them? 

There's a discussion on the same issue here: https://github.com/charettes/django-admin-enhancer/issues/23

### PR DESCRIPTION
**Pull request by rpkilby**
- Drops support for anything below Django 1.8 (current LTS release)
- ~~Cleans up some pep8 E303 warnings. I can remove this commit - just went ahead because my editor lit up like a christmas tree.~~
- ~~Resolves 183~~.
- Supersedes / closes #162.
- Supersedes / closes #204.

### CODE REVIEW

**Comment by romgar** [comment]
I'm not really convinced that we should drop support for old versions if it's not a really big issue.

I agree, in theory we will have a cleaner code, but that will prevent people that can't afford to upgrade their project to use these libraries.

**Comment by rpkilby** [comment]
> but that will prevent people that can't afford to upgrade their project to use these libraries.

There's nothing preventing them from using older versions of the package.

**Comment by romgar** [comment]
I get your point @rpkilby. 

Usually, I'm definitely dropping compatibility when I need to develop new functionalities that would be really annoying/impossible to implement in old versions with compatible code, not only for the sake of code cleaning and following Django support cycle.

@carljm what was your compatibility policy on this package until now, and your views about it?

**Comment by romgar** [comment]
After extra thoughts, I think we shouldn't encourage people to stay on old unsupported/insecure versions of Django by making the third-party apps compatible with these old versions.

@rpkilby would you mind updating your PR?

**Comment by carljm** [comment]
My policy up til now has been "drop support for Django versions when it becomes inconvienent to support them" :-) If there is more active maintenance now, I have no problem with a policy to drop support for Django versions without upstream support.

**Comment by romgar** [comment]
@rpkilby would you mind updating your PR to resolve conflicts? It would be nice to have it live!

**Comment by rpkilby** [comment]
Sure - it completely slipped my mind

**Comment by rpkilby** [comment]
The PR has been updated.

**Comment by romgar** [comment]
Great @rpkilby! I will have a look really soon!

File: model_utils/managers.py [hunk 1]
```diff
@@ -143,10 +139,10 @@ def _get_ancestors_path(self, model, levels=None):
         if levels:
             levels -= 1
         while parent_link is not None:
-            if django.VERSION < (1, 8):
-                related = parent_link.related
-            else:
+            if django.VERSION < (1, 9):
                 related = parent_link.rel
```
Comment (romgar) [inline model_utils/managers.py]: I guess this line is generating a warning in Django `1.10`?

**Comment by romgar** [comment]
You did even a bit more than dropping unsupported versions, but also fixed some deprecation warnings!

File: model_utils/managers.py (hunk 1 above)
Comment (rpkilby) [inline model_utils/managers.py]: Yep https://docs.djangoproject.com/en/1.10/releases/1.9/#field-rel-changes

**Comment by romgar** [comment]
@carljm I will merge this MR soon. It might be a good idea to do an official release with the latest version before this merge, as some fixes are already in master and people still on old versions could benefit from them.

**Comment by carljm** [comment]
@romgar 2.6.1 is now on PyPI, thanks for the nudge.

**Comment by romgar** [comment]
@rpkilby do you mind resolving the conflict with newly updated `CHANGES` file? I merge it right after.

**Comment by rpkilby** [comment]
Done - the changes have been added to the unreleased section.

**Comment by romgar** [comment]
Great job @rpkilby, thanks a lot!

**Pull request merged**